trie-db/src/proof/generate.rs

```diff
@@ -210,7 +245,7 @@ impl<'a, C: NodeCodec> StackEntry<'a, C> {
 				);
 				&mut hash.as_mut()[..encoded_child.len()].copy_from_slice(encoded_child);
```

Comment:
this would fix the warning
```suggestion
				hash.as_mut()[..encoded_child.len()].copy_from_slice(encoded_child);
```